sklearn/metrics/cluster/unsupervised.py

```diff
@@ -200,4 +200,6 @@ def silhouette_samples(X, labels, metric='euclidean', **kwds):
 
     sil_samples = inter_clust_dists - intra_clust_dists
     sil_samples /= np.maximum(intra_clust_dists, inter_clust_dists)
-    return sil_samples
+
+    # nan values are for clusters of size 1, and should be 0
```

Comment:
@MechCoder suggested testing it differently right? Could you look into that

(Pulled from the hidden diff)
> This test is not a regression test. It will pass in master.
Can you remove this and add a new test testing something like this
```
distances = np.random.RandomState(0).rand(5)
A = _intra_cluster_distance(distance, np.array([0, 1, 1, 1, 1]), 0)
assert_true(np.isnan(A))
```